drivers/vulkan/rendering_device_vulkan.cpp

```diff
@@ -5064,38 +5241,68 @@ RID RenderingDeviceVulkan::shader_create_from_bytecode(const Vector<uint8_t> &p_
 	shader->compute_local_size[1] = compute_local_size[1];
 	shader->compute_local_size[2] = compute_local_size[2];
 	shader->specialization_constants = specialization_constants;
+	shader->specialization_spirv.clear();
 	shader->name = name;
 
 	String error_text;
 
+	// TODO: What condition should we use for this to be enabled?
+	bool patch_spirv_spec_constants = true;
```

Comment:
Maybe use a project setting? The same way there's one for performing depth pre-pass or not depending on vendor. Platform suffix can also be used to have it enabled by default for some vendors on certain platforms. Something like that?

Reply:
I think we could just store the boolean on this class defaulted to false and we set it to true based on a vendor ID whitelist when fetching the physical device properties (and we only enable it on Adreno for now).

Reply:
With a project setting you can achieve that. See these in the OpenGL renderer:
https://github.com/godotengine/godot/blob/6916349697a4339216469e9bf5899b983d78db07/servers/rendering_server.cpp#L3363-L3364
https://github.com/godotengine/godot/blob/6916349697a4339216469e9bf5899b983d78db07/drivers/gles3/storage/config.cpp#L108-L123

However, I wonder if there's much value if that being user-configurable or if hard-coding it would do the trick just fine.

Reply:
I think we're better off hard-coding it, driver workarounds aren't something the user needs to be concerned about IMO.